src/dymaptic.GeoBlazor.Core/Objects/AttributesDictionary.cs

```diff
@@ -0,0 +1,313 @@
+﻿using Microsoft.AspNetCore.Components;
+using ProtoBuf;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+
+namespace dymaptic.GeoBlazor.Core.Objects;
+
+/// <summary>
+///     Dictionary of Graphic Attributes that can be asynchronously updated
+/// </summary>
+[JsonConverter(typeof(AttributesDictionaryConverter))]
+public class AttributesDictionary : IEquatable<AttributesDictionary>
+{
+    /// <summary>
+    ///     Constructor
+    /// </summary>
+    public AttributesDictionary()
+    {
+        _backingDictionary = new Dictionary<string, object>();
+    }
+
+    /// <summary>
+    ///     Constructor from existing dictionary
+    /// </summary>
+    /// <param name="dictionary">
+    ///     The dictionary to use
+    /// </param>
+    public AttributesDictionary(Dictionary<string, object> dictionary)
+    {
+        _backingDictionary = new Dictionary<string, object>();
+        var options = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
+
+        foreach (KeyValuePair<string, object> kvp in dictionary)
+        {
+            if (kvp.Value is JsonElement jsonElement)
+            {
+                object? typedValue = jsonElement.ValueKind switch
```

Comment:
had to do some research on Valuekind (https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.valuekind?view=net-7.0)
So this is what sets up the "conversion" of the attribute dictionary to allow the change comparison to occur.

Reply:
This is kludgy but yes, we're basically force-checking the type from JSON to C#.